influxdb3_write/src/write_buffer/queryable_buffer.rs

```diff
@@ -77,51 +77,41 @@ impl QueryableBuffer {
         let schema = table.schema.clone();
         let arrow_schema = schema.as_arrow();
 
-        let mut chunks: Vec<Arc<dyn QueryChunk>> = vec![];
-
         let buffer = self.buffer.read();
 
         let Some(db_buffer) = buffer.db_to_table.get(db_schema.name.as_ref()) else {
-            return Ok(chunks);
+            return Ok(vec![]);
         };
         let Some(table_buffer) = db_buffer.get(table_name) else {
-            return Ok(chunks);
+            return Ok(vec![]);
         };
 
-        let batches = table_buffer
-            .record_batches(Arc::clone(&arrow_schema), filters)
-            .map_err(|e| DataFusionError::Execution(format!("error getting batches {}", e)))?;
-
-        let timestamp_min_max = table_buffer.timestamp_min_max();
-
-        let row_count = batches.iter().map(|batch| batch.num_rows()).sum::<usize>();
-
-        let chunk_stats = create_chunk_statistics(
-            Some(row_count),
-            &schema,
-            Some(timestamp_min_max),
-            &NoColumnRanges,
-        );
-
-        chunks.push(Arc::new(BufferChunk {
-            batches,
-            schema: schema.clone(),
-            stats: Arc::new(chunk_stats),
-            partition_id: TransitionPartitionId::new(
-                TableId::new(0),
-                &PartitionKey::from("buffer_partition"),
-            ),
-            sort_key: None,
-            id: ChunkId::new(),
-            chunk_order: ChunkOrder::new(
-                chunks
-                    .len()
-                    .try_into()
-                    .expect("should never have this many chunks"),
-            ),
-        }));
-
-        Ok(chunks)
+        Ok(table_buffer
+            .partitioned_record_batches(Arc::clone(&arrow_schema), filters)
+            .map_err(|e| DataFusionError::Execution(format!("error getting batches {}", e)))?
+            .into_iter()
+            .map(|(gen_time, (ts_min_max, batches))| {
+                let row_count = batches.iter().map(|b| b.num_rows()).sum::<usize>();
+                let chunk_stats = create_chunk_statistics(
+                    Some(row_count),
+                    &schema,
+                    Some(ts_min_max),
+                    &NoColumnRanges,
+                );
+                Arc::new(BufferChunk {
+                    batches,
+                    schema: schema.clone(),
+                    stats: Arc::new(chunk_stats),
+                    partition_id: TransitionPartitionId::new(
+                        TableId::new(0),
+                        &PartitionKey::from(gen_time.to_string()),
```

Comment:
I did not see a better way to construct this `PartitionKey`, which is a Hash, instead of the original `i64`-based partition key which are being deprecated in IOx/core. Perhaps, if there is a more efficient way to do this, we can upstream some changes into IOx/core.